### PR TITLE
Allow to call `~` with `foo.~` style

### PR DESCRIPTION
**Pull request by MakeNowJust**
Currently `~` method cannot call with `foo.~` style.

```crystal
1.~
# Syntax error in foo.cr:1: expecting any of these tokens: IDENT, CONST, +, -, *, /, %, |, &, ^, **, <<, <, <=, ==, !=, =~, !~, >>, >, >=, <=>, ===, [], []=, []?, [ (not '~')
#
# 1.~
#   ^
#
```

All operators except `~` can call with such style. So I thought it is bug and fixed this.

Thank you.

### CODE REVIEW

File: src/compiler/crystal/syntax/to_s.cr [hunk 1]
```diff
@@ -367,12 +367,19 @@ module Crystal
       elsif node_obj && !letter_or_underscore?(node.name) && node.args.size == 1
         in_parenthesis(need_parens, node_obj)
 
-        @str << ' '
-        @str << decorate_call(node, node.name)
-        @str << ' '
-
         arg = node.args[0]
-        in_parenthesis(need_parens(arg), arg)
+        if node.name == "~"
```
Comment (RX14) [inline src/compiler/crystal/syntax/to_s.cr]: Why is `~` special and needs a special branch?

Comment (MakeNowJust) [reply src/compiler/crystal/syntax/to_s.cr]: `1.~(2)` is valid syntax now, but `ASTNode#to_s` does not expect this, so it returns `1 ~ 2` in this case. It is for such a case.

Comment (RX14) [reply src/compiler/crystal/syntax/to_s.cr]: I guess i'm asking is what makes `1.~(2)` special instead of `1.+(2)` that it needs to be different here

Comment (MakeNowJust) [reply src/compiler/crystal/syntax/to_s.cr]: `1 + 2` is valid, but `1 ~ 2` is invalid.

Comment (RX14) [reply src/compiler/crystal/syntax/to_s.cr]: ah, makes sense

**Pull request merged**